### ISSUE TITLE
Deployments with Custom Python Packages

### ISSUE THREAD

**Issue by FeryET** [PrefectHQ/prefect#9937]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

There is no way to use a preinstalled python package in an environment for deployments. The source code must exist in a block or in a local directory for a prefect deployment to find it.

### Describe the proposed behavior

Since many will create their own worker types, or might develop their own flow packages for better versioning, reliability and ease of installment of dependencies, using prefect workers/agents in a virtualenv in which the packages are pre-installed, or installing them using a hook or even as a `pypi storage block ` will be great.

### Example Use

Currently I am developing a system in which I'm using prefect as an orchestrator to have observable/reliable pipelines. The code is highly dynamic and the dependencies are changing rapidly. To make it possible to use a preinstalled package in my environment, I am currently tightly coupling the `Deployment.path` field of my `Deployment` objects with a pre-defined location in my worker images that will execute my flows.

This is highly error prone, and since workers don't support storage blocks this is the only way to use them right now for me.

**Comment by desertaxle** [comment]
Thanks for the issue @FeryET! This would involve expanding the entrypoints prefect accepts on a deployment to include a module path (e.g., `module.path.to.flow`). We have utilities in `prefect. utilities.importutils` that will help with importing from a module path.

**Comment by donjpierce** [comment]
@FeryET 

Great find. I believe I am having this issue as well. Do you mind if I describe my issue to you to confirm?

I have built and installed a custom python package (hosted on github) to re-use code across different prefect flows in my deployment. I then install the custom package along with other pypi packages in my prefect deployment's virtual environment. Everything imports and runs fine in the docker container I have built. Local flow execution also works end-to-end. However Prefect cloud does not recognize my custom package as a package, and all my flows fail when I import the custom package in production, even though I am using the same docker image.

**Comment by FeryET** [comment]
> @FeryET 
> 
> Great find. I believe I am having this issue as well. Do you mind if I describe my issue to you to confirm?
> 
> I have built and installed a custom python package (hosted on github) to re-use code across different prefect flows in my deployment. I then install the custom package along with other pypi packages in my prefect deployment's virtual environment. Everything imports and runs fine in the docker container I have built. Local flow execution also works end-to-end. However Prefect cloud does not recognize my custom package as a package, and all my flows fail when I import the custom package in production, even though I am using the same docker image.
> 

Yes this is the same issue I'm having. I've proposed a small fix [in this pull requsst ](https://github.com/PrefectHQ/prefect/pull/9987). Tried it on a local server and it worked. See if this alleviates your issue?

**Comment by petri** [comment]
Would be very useful addition. Much better to deploy packaged flows to/from private pypi than give GitHub access, and when building containers is an overkill.
